### ISSUE TITLE
Where will end to end testing for extensions be performed?

### ISSUE THREAD

**Issue by peternied** [opensearch-project/opensearch-sdk-java#634]
For the overall Extensions effort, there should be test cases that verify functionality end to end.  While many of these case can be tested in isolation integration testing required to ensure backwards compatibility / confirm breaking changes.

In order to run these scenarios changes from OpenSearch and OpenSearch SDK + Security Plugin will need to be collected and configured for the test environment.  Where should this testing be performed, this repository?

**Comment by peternied** [comment]
If I might offer an implementation suggestion, the security test require considerable setup it might be worthwhile to see how this is done to create flexible and readable test cases, see [PrivilegesEvaluatorTest.java](https://github.com/opensearch-project/security/blob/a9bad908d6692be439694288958bbb4b42f4b577/src/integrationTest/java/org/opensearch/security/privileges/PrivilegesEvaluatorTest.java) that uses composition of the cluster state.

**Comment by minalsha** [comment]
@saratvemulapalli , @ryanbogan thoughts?

**Comment by ryanbogan** [comment]
I'm working on an integration test framework for extensions that will modify the OpenSearch gradle plugin so that both extensions and plugins can be used at the same time.  How many of the security features for extensions are going to be required/optional?

**Comment by peternied** [comment]
@ryanbogan I don't think this is a question of 'security' features to be tested, but how would we be sure that extensions are working correctly if as we update the several dependent codebases (OpenSearch / SDK / Security / JobScheduler?).  While security features have a good chance to break functionality, there are many features such as job scheduler that can also be broken at integration time that should be validated.

To be cautious ever feature would have an integration test and how many scenarios are tests would be a separate exercise on a case-by-case basis.

I'm curious about that system you are thinking about, do you think it would be able to cover these broad kinds of configurations, what kind of trade-offs or limitations are you keeping in mind?  Would you want to write up a proposal in this issue or a new separate issue?

**Comment by dblock** [comment]
As an author of an extension I'd like to be able to easily declare a test matrix that uses any build of OpenSearch to test by extension in CI/CD, aka mostly GitHub actions.

A couple of thoughts: 1) the gradle test system of OpenSearch for plugins is very powerful 2) the gradle test system of OpenSearch for plugins is a kitchen sink. Whichever way we go I would not inherit anything from OpenSearch core to avoid having to drag the kitchen sink into the development environments.
